### PR TITLE
Fix min-chained-call-depth rule

### PR DESCRIPTION
**Pull request by georgekaran**
## Summary

This pull request aims to fix some conflicts between `min-chained-call-depth` and `newline-per-chained-call`.

* Added a new option to the `min-chained-call-depth` called `ignoreChainWithDepth`, which has on purpose the same name as in the`newline-per-chained-call`.

### Examples 
✅ Good
```ts
Array(10)
    .fill(10)
    .map(foo => foo)
    .slice(5);
```
```ts
Array(10).fill(10);
```

❌ Bad
```ts
Array(10)
    .map(foo => foo)
    .slice(5);
```
```ts
Array(10)
    .fill(10);
```

### Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Review by Fryuni — APPROVED** [review]

File: src/rules/min-chained-call-depth/index.ts [hunk 1]
```diff
@@ -104,13 +114,33 @@ export const minChainedCallDepth = createRule({
...
             // We only inline the first level of chained calls.
             // If the current call is nested inside another call, we can skip.
-            if (getDepth(callee) > 1) {
+            if (currentDepth > 1) {
+                return;
+            }
+
+            const {maxLineLength = 100, ignoreChainWithDepth = 2} = context.options[0] ?? {};
+
+            // If the max depth is greater than ignoreChainWithDepth, we can skip.
```
Comment (marcospassos) [inline src/rules/min-chained-call-depth/index.ts:129]:
```suggestion
            // If the max depth is greater than ignoreChainWithDepth, skip.
```

File: src/rules/min-chained-call-depth/index.ts [hunk 2]
```diff
@@ -25,6 +25,12 @@ export const minChainedCallDepth = createRule({
                         minimum: 1,
                         default: 100,
                     },
+                    ignoreChainWithDepth: {
```
Comment (marcospassos) [inline src/rules/min-chained-call-depth/index.ts:28]:
```suggestion
                    ignoreChainDeeperThan: {
```

File: src/rules/min-chained-call-depth/index.ts [hunk 3]
```diff
@@ -104,13 +114,33 @@ export const minChainedCallDepth = createRule({
                 return;
             }
 
+            const currentDepth = getDepth(callee);
+
+            maxDepth = Math.max(maxDepth, currentDepth);
+
             // We only inline the first level of chained calls.
```
Comment (marcospassos) [inline src/rules/min-chained-call-depth/index.ts:121]:
```suggestion
            // Only affect the root level as the total depth is must be known.
```

File: src/rules/min-chained-call-depth/index.ts (hunk 1 above)
Comment (marcospassos) [inline src/rules/min-chained-call-depth/index.ts:129]:
```suggestion
            // If the max depth is greater than ignore threshold, skit
```

File: src/rules/min-chained-call-depth/index.ts [hunk 4]
```diff
@@ -104,13 +114,33 @@ export const minChainedCallDepth = createRule({
...
+            // If the max depth is greater than ignoreChainWithDepth, we can skip.
+            //
+            // Example:
+            //     ```ts
+            //     Array(10)
+            //         .fill(0)
+            //         .map(x => x + 1)
+            //         .slice(0, 5);
+            //     ```
+            //     In this case the depth is 3, and the default value of ignoreChainWithDepth is 2.
```
Comment (marcospassos) [inline src/rules/min-chained-call-depth/index.ts:138]: update the name

**Review by marcospassos — CHANGES_REQUESTED** [review]

File: docs/rules/min-chained-call-depth.md [hunk 5]
```diff
@@ -74,6 +74,27 @@ expect(screen.getElementById("very-long-identifier"))
...
+```jsx
+Array(10)
+    .fill(0)
+    .map(foo => foo);
+```
+
+#### ✅ Correct
+
+```jsx
+/*eslint min-chained-call-depth: ["error", {"ignoreChainDeeperThan": 1}]*/
```
Comment (marcospassos) [inline docs/rules/min-chained-call-depth.md:92]:
```suggestion
/* eslint min-chained-call-depth: ["error", {"ignoreChainDeeperThan": 1}] */
```

File: src/rules/min-chained-call-depth/index.ts [hunk 6]
```diff
@@ -88,29 +98,49 @@ export const minChainedCallDepth = createRule({
...
-                // If the callee is not a member expression, we can skip.
+                // If the callee is not a member expression, skip.
                 // For example, root level calls like `foo();`.
                 callee.type !== AST_NODE_TYPES.MemberExpression
-                // If the callee is a computed member expression, like `foo[bar]()`, we can skip.
+                // If the callee is a computed member expression, like `foo[bar]()`, skip.
                 || callee.computed
                 /* eslint-disable-next-line @typescript-eslint/ban-ts-comment --
                 * NewExpression is a possible callee object type
                 */
```
Comment (marcospassos) [inline src/rules/min-chained-call-depth/index.ts:108]:
```suggestion
                /* eslint-disable-next-line @typescript-eslint/ban-ts-comment --
                 * NewExpression is a possible callee object type
                 */
```

**Review by marcospassos — CHANGES_REQUESTED** [review]

**Pull request merged**